File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> allnums;
        vector<int> results;

        for(int i = 0; i < nums.size(); i++){

            if(allnums.count(target - nums[i])){
                results.push_back(i);
                results.push_back(allnums[target - nums[i]]);
                return results;
            }

            allnums[nums[i]] = i;
        }
        
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        answer = []
        my_dict = {}



        for i, n in enumerate(nums):
            if target - n in my_dict:
                answer.append(i)
                answer.append(my_dict.get(target-n))
                return answer

            my_dict[n] = i
        
        return answer
